tests/components/nest/device_info_test.py

```diff
@@ -0,0 +1,103 @@
+"""Test for properties for devices common to all entity types."""
+
+from google_nest_sdm.device import Device
+
+from homeassistant.components.nest.device_info import DeviceInfo
+
+
+def test_device_custom_name():
+    """Test a device name from an Info trait."""
+    device = Device.MakeDevice(
+        {
+            "name": "some-device-id",
+            "type": "sdm.devices.types.DOORBELL",
+            "traits": {
+                "sdm.devices.traits.Info": {
+                    "customName": "My Doorbell",
+                },
+            },
+        },
+        auth=None,
+    )
+
+    device_info = DeviceInfo(device)
+    assert device_info.device_name == "My Doorbell"
+    assert device_info.device_model == "Doorbell"
+    assert device_info.device_brand == "Google Nest"
+    assert device_info.device_info == {
+        "identifiers": {("nest", "some-device-id")},
+        "name": "My Doorbell",
+        "manufacturer": "Google Nest",
+        "model": "Doorbell",
+    }
+
+
+def test_device_name_room():
+    """Test a device name from the room name."""
+    device = Device.MakeDevice(
+        {
+            "name": "some-device-id",
+            "type": "sdm.devices.types.DOORBELL",
+            "parentRelations": [
+                {"parent": "some-structure-id", "displayName": "Some Room"}
+            ],
+        },
+        auth=None,
+    )
+
+    device_info = DeviceInfo(device)
+    assert device_info.device_name == "Some Room"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6372789)